src/content/platform-integration/web/embedding-flutter-web.md

```diff
@@ -0,0 +1,326 @@
+---
+title: Embedding Flutter on the Web
+short-title: Embedding Flutter web
+description: Learn the different ways to embed Flutter views into web content.
+---
+
+Flutter views and web content can be composed to produce a web application
+in different ways. Choose one of the following depending on your use-case:
+
+* A Flutter view controls the full page (full-screen mode)
+* Adding Flutter views to an existing web application (embedded mode)
+
+## Full-screen mode
+
+In full screen mode, the Flutter web application takes control of the whole
+browser window and covers its viewport completely when rendering. This is the
+default embedding mode for Flutter, and no additional configuration is needed.
+
+```html highlightLines=6
+<!DOCTYPE html>
+<html>
+  <head>
+  </head>
+  <body>
+    <script src="flutter_bootstrap.js" defer></script>
+  </body>
+</html>
+```
+
+When Flutter web is bootstrapped without referencing `multiViewEnabled` or a 
+`hostElement`, it uses full-screen mode.
+
+See [Customize app initialization][] to learn more about the
+`flutter_bootstrap.js` file.
+
+[Customize app initialization]: {{site.docs}}/platform-integration/web/initialization/
+
+### `iframe` embedding
+
+Full-screen mode is recommended when embedding a Flutter web application in an
+`iframe`. The page that embeds the `iframe` can size and position it as needed,
+and Flutter will fill it completely.
+
+```html
+<iframe src="https://url-to-your-flutter/index.html"></iframe>
+```
+
+Check [`<iframe>`: The Inline Frame element][] in MDN to learn more about the
+pros and cons of `iframe`s.
+
+[`<iframe>`: The Inline Frame element]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe
+
+## Embedded mode
+
+Flutter web applications can also render content into an arbitrary number of
+elements (commonly `div`s) of another web application; this is called "embedded
+mode" (or "multi-view").
+
+In this mode:
+
+* A Flutter web application can launch, but doesn't render until the first
+"view" is added, with `addView`.
+* The host application may add or remove views from the embedded Flutter web
+application.
+* The Flutter application gets notified of when views are added or removed, so
+it can adjust its widgets accordingly.
+
+### Enable multi-view mode
+
+Enable multi-view mode setting `multiViewEnabled: true` in the
+`initializeEngine` method as shown:
+
+```js highlightLines=8
+// flutter_bootstrap.js
+{% raw %}{{flutter_js}}{% endraw %}
+{% raw %}{{flutter_build_config}}{% endraw %}
+
+_flutter.loader.load({
+  onEntrypointLoaded: async function onEntrypointLoaded(engineInitializer) {
+    let engine = await engineInitializer.initializeEngine({
+      multiViewEnabled: true, // Enables embedded mode.
+    });
+    let app = await engine.runApp();
+    // Make this `app` object available to your JS app.
```

Comment:
How does the developer use `app` exactly? It's not very useful if it's only available in the scope of the `onEntrypointLoaded` callback. It seems like the developer needs to do something like:

```javascript
function getApp() {
  return new Promise((resolve) => { 
    _flutter.loader.load({
      onEntrypointLoaded: async function onEntrypointLoaded(engineInitializer) {
        let engine = await engineInitializer.initializeEngine({
          multiViewEnabled: true, 
        });
        let app = await engine.runApp();
        resolve(app); // Resolve the promise with the app
      }
    });
  });
}
```

Reply:
Isn't this the same, without needing a promise (that can't be re-resolved upon hot reload?):

```js
{{flutter_js}}
{{flutter_build_config}}

_flutter.loader.load({
    onEntrypointLoaded: async function onEntrypointLoaded(engineInitializer) {
        let engine = await engineInitializer.initializeEngine({
            multiViewEnabled: true,
        });
        let app = await engine.runApp();
        doThingsWithApp(app); // Pass the app to the JS that needs it.
    }
});

function doThingsWithApp(app) {
    // var app = await getApp(); // app comes as a parameter now
    let viewId1 = app.addView({
        hostElement: document.querySelector('div#host1')
    })
    let viewId2 = app.addView({
        hostElement: document.querySelector('div#host2')
    })
    console.log('view id = ' + viewId1)
    console.log('view id 2 = ' + viewId2)
}
```

I'm not sure how prescriptive we want to be with this bit of the code; it's highly dependent on the host web app, isn't it?

Reply:
That's also a reasonable approach. I'm OK to keep this snippet as-is since it's up to the developer to decide how they want to access `app`.